src/components/common/FavoriteButton.tsx

```diff
@@ -70,7 +70,7 @@ const FavoriteButton: FC<Props> = ({ item, size, type, onClick }) => {
       handleUnfavorite={handleUnfavorite}
       handleFavorite={handleFavorite}
       tooltip={text}
-      type={type}
+      type={type as ActionButtonVariant}
```

Comment:
Is this caused by the nullable? If yes, consider using a non-nullable prop type for `type`, or returning `null` if not set (but I think removing the nullable makes sense).